### ISSUE TITLE
Picker.visible isn't updating within script?

### ISSUE THREAD

**Issue by evolross** [tovic/color-picker#40]
If I call `picker.enter()` and the color picker opens successfully, but I then go to check the visibility of the color picker using:
```
CP.__instance__["myColorPickerId"].visible
```
In code, `visible` never updates to true.  But if I check the value using `console.log` it does update?

I have an event handler for a button-click that I want to open the color picker upon first click, then the second click will save the value and close the color picker.  But the `visibility` flag never updates, even after fetching the color picker again.

My code organization is a little different than your examples.  But it should still work.  Checking `CP.__instance__` should already have fresh values right?

Does this have something to do with `CP.__instance__[...]` returning an anonymous function?

**Comment by tovic** [comment]
Did your code creates new instance from the same element ID on every click button?

For simple clue, here&rsquo;s how I handle the visibility attribute of this plugin. It works by checking the `parentNode` attribute value of the `self` node.

1. https://github.com/tovic/color-picker/blob/1722fec9b8564eaf3b56b00a2b608d29ce129087/color-picker.js#L587
2. https://github.com/tovic/color-picker/blob/1722fec9b8564eaf3b56b00a2b608d29ce129087/color-picker.js#L379-L382
3. https://github.com/tovic/color-picker/blob/1722fec9b8564eaf3b56b00a2b608d29ce129087/color-picker.js#L387
4. https://github.com/tovic/color-picker/blob/1722fec9b8564eaf3b56b00a2b608d29ce129087/color-picker.js#L436

So, if something makes this method not work for you, you can always use `if (CP.__instance__.foo.parentNode)` in place of `if (CP.__instance__.foo.visible)`.

**Comment by evolross** [comment]
Yes, every time my event fires I refetch the color picker like so:
```
'click .color-select-custom:enabled': function(event) {
    let answerId = this._id;
    let $colorPicker = CP.__instance__["colorPickerText_" + answerId];

    console.log("$colorPicker.visible: ", $colorPicker.visible);  // Always returns false

   if(!$colorPicker.visible)
      $colorPicker.enter();  // This works, so I have the correct color picker
   else {
       ...  //  This never gets reached
   }
}
```
The above code works fine. I can get the color picker and because `visible` is false to begin with, it opens the color picker fine.  But upon calling this event again, `visible` always returns false.

The idea is there's a button that opens the color picker if it's not already open and then saves the value if it is open.

Yet out in the Chrome console, I can run `CP.__instance__["colorPickerText_theId"].visible` and `visible` returns true after clicking the button the first time and having `enter()` called.

It's just not updating in my event code when it reruns even though I am fetching it again.  I resorted to adding my own `data-attribute` flag on the HTML element to know if it's open or not.  Check this works.  It might be some kind of scope issue but I've never had something like this happen before.  Most JS objects and plugins update and work correctly like this.

**Comment by evolross** [comment]
I figured out the issue. A little tricky but it was a mistake on my part.

I was running color picker in default mode which uses `click` as the event to open it.  This opened the color picker nicely with a click on the text input.  My issue was I was also using an independent button outside of the color picker to open _and_ to close/save it. The same button actually. Basically the first click opens the color picker. The second click saves and closes the opened color picker.

So when that button went to save and close the color picker, because the default `click` mode of the color picker is to close the color picker if a `click` occurs outside of the color picker, the color picker was calling `exit()` _before_ my button's `click` handler that tries to save and close.  So that's why `visibility` was always false.  I couldn't see this because that same button, if the color picker isn't visible, calls `enter()` to show the color picker.  It happened so fast I couldn't see the color picker hiding then showing.

It was only after logging the `enter` and `exit` events that I realized what was happening. I could see the `exit` firing before my button's `click` event.

To get the functionality I want, I changed the color picker to start disabled by passing `false` to the constructor so clicking outside of the color picker doesn't automatically close it. This works but I have to add an event to the text input to open the color picker on a `click` (because I want the color picker to open this way too) and I had to add a click event handler for the area _outside_ of the color picker and the button to call `exit()` on the color picker to "auto hide" it on a click away.

I tried it the other way with keeping the color picker constructed using the default `click` event to keep the existing functionality, but no matter what I did, the `exit()` got called before my buttons' `click` handlers. No matter what, `visibility` was always false.  I couldn't _override_ your plugin's handling of clicking outside the color picker, including any buttons, from calling `exit()` on the color picker.

It would be cool to add a constructor parameter to maybe enable/disable this and/or pass in buttons that won't trigger the "click away" functionality in the default mode. 

Whew... spent a lot of time on this one.
